### PR TITLE
[Feature]use shard group managing shards

### PR DESCRIPTION
**Pull request by kevincai**
* update staros jar version to 0.2.0

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

## What type of PR is this：
- [ ] BugFix
- [X] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 107 / 145 (73.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|2|00.00%|[171, 176]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[866]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|6|50.00%|[745, 805, 1313]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|25|40|62.50%|[291, 292, 334, 338, 339, 340, 341, 342, 346, 348, 349, 350, 351, 352, 353]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|4|75.00%|[568]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|49|62|79.03%|[55, 80, 83, 85, 86, 87, 97, 115, 121, 149, 168, 170, 171]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|13|16|81.25%|[1306, 1576, 1580]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|4|4|100.00%|[]|

**Review by imay — DISMISSED** [review]

**Review by tracymacding — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by kevincai** [comment]
has compatibility issue with old version, need look into it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 107 / 142 (75.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|2|00.00%|[171, 176]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[866]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|25|40|62.50%|[291, 292, 334, 338, 339, 340, 341, 342, 346, 348, 349, 350, 351, 352, 353]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|4|75.00%|[568]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|49|62|79.03%|[55, 80, 83, 85, 86, 87, 97, 115, 121, 149, 168, 170, 171]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|13|16|81.25%|[1306, 1576, 1580]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -1099,7 +1097,6 @@ private List<Partition> createPartitionList(Database db, OlapTable copiedTable,
                     createPartition(db, copiedTable, partitionId, partitionName, version, tabletIdSet);
 
             partitionList.add(partition);
-            tabletIdSetForAll.addAll(tabletIdSet);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1102]: `tabletIdSetForAll` is used to delete tablet from inverted index when create partition failed.

**Review by tracymacding — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -1553,12 +1545,7 @@ public void dropPartition(Database db, Table table, DropPartitionClause clause)
                     }
                 }
             }
-            tabletIdSet = olapTable.dropPartition(db.getId(), partitionName, clause.isForceDrop());
-
-            if (olapTable instanceof MaterializedView) {
-                MaterializedView mv = (MaterializedView) olapTable;
-                SyncPartitionUtils.dropBaseVersionMeta(mv, partitionName);
-            }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1561]: why remove materialized view code?

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 3]
```diff
@@ -1577,27 +1564,20 @@ public void dropPartition(Database db, Table table, DropPartitionClause clause)
...
                 clause.isForceDrop());
     }
 
     public void replayDropPartition(DropPartitionInfo info) {
         Database db = this.getDb(info.getDbId());
         db.writeLock();
-        Set<Long> tabletIdSet = new HashSet<Long>();
         try {
             OlapTable olapTable = (OlapTable) db.getTable(info.getTableId());
+            Partition partition = olapTable.getPartition(info.getPartitionName());
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1576]: not used.

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 4]
```diff
@@ -4511,18 +4488,6 @@ public void truncateTable(TruncateTableStmt truncateTableStmt) throws DdlExcepti
                 tblRef.getName().toSql(), tblRef.getPartitionNames());
     }
 
-    private void deleteUselessTabletAndShard(Set<Long> tabletIdSet, OlapTable olapTable) {
-        // create partition failed, remove all newly created tablets
-        for (Long tabletId : tabletIdSet) {
-            GlobalStateMgr.getCurrentInvertedIndex().deleteTablet(tabletId);
-        }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4518]: it is necessary to delete tablet from inverted index when failed.

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 5]
```diff
@@ -1766,6 +1766,12 @@ public class Config extends ConfigBase {
     @ConfField
     public static String starmgr_s3_sk = "";
 
+    /**
+     * empty shard group clean threshold (by create time).
+     */
+    @ConfField
+    public static long shard_group_clean_threshold_ms = 3600000L;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1773]: better to use second as the unit.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 105 / 141 (74.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|2|00.00%|[171, 176]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[866]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|25|40|62.50%|[291, 292, 334, 338, 339, 340, 341, 342, 346, 348, 349, 350, 351, 352, 353]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|11|15|73.33%|[1310, 1587, 4417, 4487]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|4|75.00%|[568]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|49|62|79.03%|[55, 80, 83, 85, 86, 87, 97, 115, 121, 149, 168, 170, 171]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 107 / 143 (74.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|2|00.00%|[171, 176]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[866]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|25|40|62.50%|[291, 292, 334, 338, 339, 340, 341, 342, 346, 348, 349, 350, 351, 352, 353]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|11|15|73.33%|[1310, 1587, 4417, 4487]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|4|75.00%|[568]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|50|63|79.37%|[59, 84, 87, 89, 90, 91, 101, 119, 125, 153, 172, 174, 175]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|

**Review by wyb — DISMISSED** [review]

**Review by tracymacding — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by sduzh — DISMISSED** [review]

**Comment by kevincai** [comment]
code rebased to latest main HEAD

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 107 / 143 (74.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|2|00.00%|[171, 176]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[866]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|25|40|62.50%|[291, 292, 334, 338, 339, 340, 341, 342, 346, 348, 349, 350, 351, 352, 353]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|11|15|73.33%|[1310, 1587, 4417, 4487]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|4|75.00%|[568]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|50|63|79.37%|[59, 84, 87, 89, 90, 91, 101, 119, 125, 153, 172, 174, 175]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**